ibis/common/grounds.py

```diff
@@ -235,21 +277,21 @@ def __reduce__(self):
         state = dict(zip(self.__argnames__, self.__args__))
         return (self.__recreate__, (state,))
 
-    def __hash__(self):
+    def __hash__(self) -> int:
         return self.__precomputed_hash__
 
-    def __equals__(self, other):
+    def __equals__(self, other) -> bool:
         return self.__args__ == other.__args__
 
     @property
     def args(self):
         return self.__args__
 
     @property
-    def argnames(self):
+    def argnames(self) -> Tuple[str, ...]:
```

Comment:
```suggestion
    def argnames(self) -> tuple[str, ...]:
```

Reply:
The whole file uses `Tuple` so it would be preferable to change them all at once. Deferring it to a follow-up.